chpl/chpl-service/src/main/java/gov/healthit/chpl/attestation/domain/AttestationPeriod.java

```diff
@@ -0,0 +1,49 @@
+package gov.healthit.chpl.attestation.domain;
+
+import java.time.LocalDate;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import gov.healthit.chpl.attestation.entity.AttestationPeriodEntity;
+import gov.healthit.chpl.util.LocalDateDeserializer;
+import gov.healthit.chpl.util.LocalDateSerializer;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class AttestationPeriod {
+    private Long id;
+
+    @JsonDeserialize(using = LocalDateDeserializer.class)
+    @JsonSerialize(using = LocalDateSerializer.class)
+    private LocalDate periodStart;
+
+    @JsonDeserialize(using = LocalDateDeserializer.class)
+    @JsonSerialize(using = LocalDateSerializer.class)
+    private LocalDate periodEnd;
+
+    @JsonDeserialize(using = LocalDateDeserializer.class)
+    @JsonSerialize(using = LocalDateSerializer.class)
+    private LocalDate submissionStart;
+
+    @JsonDeserialize(using = LocalDateDeserializer.class)
+    @JsonSerialize(using = LocalDateSerializer.class)
+    private LocalDate submissionEnd;
+
+    private String description;
+
+    public AttestationPeriod(AttestationPeriodEntity entity) {
```

Comment:
Should we have a discussion about where we are putting the domain object creation code? I have been liking thinking of the domain objects as things that could be in their own jar and handed off to somebody (they aren't right now in reality due to dependencies on DTOs and entities). So I have been putting methods in the entity classes like "toDomain()" that uses the entity instance to create the domain object. And we clearly are thinking opposite ways about that since the domain objects here are referencing the entity. I don't care for now but would like to agree on what to do moving forward.